core/nodejsActionBase/prelauncher.js

```diff
@@ -0,0 +1,113 @@
+#!/usr/local/bin/node
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+try {
+  const readline = require('readline');
+  const fs = require("fs")
+
+  const { NodeActionRunner, initializeActionHandler } = require('/nodejsAction/runner');
+  const NodeActionLambdaRunner = (() => {
+    try {
+        let lambda = require('/nodejsAction/lambda');
+        return lambda;
+    } catch (e) {}
+  })();
+
+  const lambdaCompat = process.env.__OW_LAMBDA_COMPAT === undefined ? false : process.env.__OW_LAMBDA_COMPAT.toLowerCase() === 'true' && NodeActionLambdaRunner !== undefined;
+
+  function doInit(message) {
```

Comment:
This is an almost 1:1 copy of the respective function as it was in the Node.js server.